File: src/iPad/MyVoteLV/MyVoteLV/Poll.m
//
//  Poll.m
//  MyVoteLV
//
//  Created by Kasey Schindler on 2/22/13.
//  Copyright (c) 2013 Magenic. All rights reserved.
//

#import "Poll.h"

@implementation Poll

- (instancetype)initPollWithDictionary:(NSDictionary *)dict
{
    self = [super init];
    
    if (self) {
        // custom setup here
        NSNumber *holder = [dict objectForKey:@"PollAdminRemovedFlag"];
        self.PollAdminRemovedFlag = holder.boolValue;
        
        holder = [dict objectForKey:@"PollDeleteFlag"];
        self.PollDeletedFlag = holder.boolValue;
        
        holder = [dict objectForKey:@"PollID"];
        self.PollID = [CommonMethods validValueForJSONNumber:holder];
        
        holder = [dict objectForKey:@"PollMaxAnswers"];
        self.PollMaxAnswers = [CommonMethods validValueForJSONNumber:holder];
        
        holder = [dict objectForKey:@"PollMinAnswers"];
        self.PollMinAnswers = [CommonMethods validValueForJSONNumber:holder];
        
        holder = [dict objectForKey:@"UserID"];
        self.UserID = [CommonMethods validValueForJSONNumber:holder];

        self.pollCategory = [Enums translateStringToPollCategoryType:[dict objectForKey:@"PollCategoryID"]];
        self.PollDateRemoved = [dict objectForKey:@"PollDateRemoved"];
        self.PollDeletedDate = [dict objectForKey:@"PollDeletedDate"];
        self.PollEndDate = [dict objectForKey:@"PollEndDate"];
        
        self.PollImageLink = [dict objectForKey:@"PollImageLink"];
        self.PollOptions = [dict objectForKey:@"PollOptions"];
        self.PollQuestion = [dict objectForKey:@"PollQuestion"];
        self.PollStartDate = [dict objectForKey:@"PollStartDate"];
    }
    
    return self;
}

@end

File: testAutomation/TestAutomation/Models/WebService/Poll.cs
using System;
using System.Collections.Generic;

namespace Models.WebService
{
	public sealed class Poll
	{
        public Poll(int? pollID, int userID, int pollCategoryID, string pollQuestion, string pollImageLink, short pollMaxAnswers, short pollMinAnswers, DateTime? pollStartDate, DateTime? pollEndDate, bool pollAdminRemovedFlag, DateTime? pollDateRemoved, bool pollDeletedFlag, DateTime? pollDeletedDate, string pollDescription, List<PollOption> pollOptions)
        {
            this.PollID = pollID;
            this.UserID = userID;
            this.PollCategoryID = pollCategoryID;
            this.PollQuestion = pollQuestion;
            this.PollImageLink = pollImageLink;
            this.PollMaxAnswers = pollMaxAnswers;
            this.PollMinAnswers = pollMinAnswers;
            this.PollStartDate = pollStartDate;
            this.PollEndDate = pollEndDate;
            this.PollAdminRemovedFlag = pollAdminRemovedFlag;
            this.PollDateRemoved = pollDateRemoved;
            this.PollDeletedFlag = pollDeletedFlag;
            this.PollDeletedDate = pollDeletedDate;
            this.PollDescription = pollDescription;
            this.PollOptions = pollOptions;
        }

        public int? PollID { get; set; }
		public int UserID { get; set; }
		public int PollCategoryID { get; set; }
		public string PollQuestion { get; set; }
		public string PollImageLink { get; set; }
		public short PollMaxAnswers { get; set; }
		public short PollMinAnswers { get; set; }
		public DateTime? PollStartDate { get; set; }
		public DateTime? PollEndDate { get; set; }
		public bool PollAdminRemovedFlag { get; set; }
		public DateTime? PollDateRemoved { get; set; }
		public bool PollDeletedFlag { get; set; }
		public DateTime? PollDeletedDate { get; set; }
		public string PollDescription { get; set; }
		public List<PollOption> PollOptions { get; set; }
	}
}